frontend/packages/console-plugin-sdk/src/typings/nav-items.ts

```diff
@@ -59,6 +67,15 @@ export const isResourceClusterNavItem = (e: Extension): e is ResourceClusterNavI
   return e.type === 'NavItem/ResourceCluster';
 };
 
+export const isSeparatorNavItem = (e: Extension): e is SeparatorNavItem => {
+  return e.type === 'NavItem/Separator';
+};
+
 export const isNavItem = (e: Extension): e is NavItem => {
```

Comment:
Not for this PR, but shouldn't we simply check e.type includes/startswith NavItem for this check instead of calling separate functions?

Reply:
Yeah, I'm not sure what the original rational was.

Reply:
This pattern is used for typescript type guards. If you need to know if it's a separator then you need to check the entire type. Then the magic comes with the return type of `is SeparatorNavItem` where it informs typescript type checked of the new type and can give you proper typing after the guard is checked.